app/org/sagebionetworks/bridge/services/AuthenticationService.java

```diff
@@ -304,7 +298,8 @@ private UserSession getSessionFromAccount(Study study, CriteriaContext context,
         if (participant.getLanguages().isEmpty() && !context.getLanguages().isEmpty()) {
             participant = new StudyParticipant.Builder().copyOf(participant)
                     .withLanguages(context.getLanguages()).build();
-            optionsService.setOrderedStringSet(study, account.getHealthCode(), LANGUAGES, context.getLanguages());
+            account.setLanguages(context.getLanguages());
+            accountDao.updateAccount(account, false);
```

Comment:
I though here to avoid loading the account one more time. During account load we "go back to the well" in two places to update the account before returning it. We do that if it's missing a health code (presumably rare, I only found one account without a health code during migration), and then again to rotate the auth token. You are right that using the editAccount function here would ignore the version of the account at this point in the code, though I believe now it is correct after fixing an integration test by transferring the version to the Account object after the reauth update. Do you still want to change to editAccount()?